lib/xapi-stdext-pervasives/pervasiveext.ml

```diff
@@ -58,16 +55,33 @@ let reraise_if bool fct =
 let ignore_exn fct = try fct () with _ -> ()
 
 (* non polymorphic ignore function *)
-let ignore_int v = let (_: int) = v in ()
-let ignore_int64 v = let (_: int64) = v in ()
-let ignore_int32 v = let (_: int32) = v in ()
-let ignore_string v = let (_: string) = v in ()
-let ignore_float v = let (_: float) = v in ()
-let ignore_bool v = let (_: bool) = v in ()
+let ignore_int v =
+  let (_ : int) = v in
+  ()
+
+let ignore_int64 v =
+  let (_ : int64) = v in
+  ()
+
+let ignore_int32 v =
+  let (_ : int32) = v in
+  ()
+
+let ignore_string v =
+  let (_ : string) = v in
+  ()
+
+let ignore_float v =
+  let (_ : float) = v in
+  ()
+
+let ignore_bool v =
+  let (_ : bool) = v in
+  ()
 
 (* To avoid some parens: *)
 (* composition of functions: *)
-let (++) f g x = f (g x)
+let ( ++ ) f g x = f (g x)
 
 (* and application *)
-let ($) f a = f a
+let ( $ ) f a = f a
```

Comment:
I don't like this one because it is not right-associative as it needs to be. That is why we use `@@`. Maybe deprecate it? (I don't recall it being used)

Reply:
I's being used for term options. I think:

```
$ rg -tml -F ' $ ' -c | cut -f 1 -d / | sort | uniq
camldm
ezlvm
ezxenstore
luna-rossa
message-switch
nbd
ocaml-qmp
ocaml-rpc-legacy
ocaml-xen-lowlevel-libs
ocaml-xenstore-xen
rrd-transport
sm-cli
stdext
testarossa
varstored-guard
vhd-tool
vncproxy
vxs
xapi-clusterd
xapi-nbd
xapi-storage
xapi-storage-script
xcp-idl
xcp-inventory
xcp-rrdd
xen-api
xenops-cli
xenopsd
xenvm
```

Reply:
for example:
```
message-switch/switch/switch_main.ml
71:    Term.(pure make $ path $ pidfile $ configfile $ statedir)
357:  (Term.(ret (pure main $ Config.term)), Term.info "main" ~doc ~man)
```

Reply:
nevermind, that is from the cmdliner package

Reply:
I know that cmdliner defines `($)` and I assume in that specific context associativity doesn't matter but I'm a bit surprised by it.